experimental/PropertyDDS/packages/property-dds/src/propertyTree.ts

```diff
@@ -878,6 +907,26 @@ export class SharedPropertyTree extends SharedObject {
 		return true;
 	}
 
+	protected reSubmitCore(content: any, localOpMetadata: unknown) {
+		// We have to provide our own implementation of the resubmit core function, to
+		// handle the case where an operation is no longer referencing a commit within
+		// the collaboration window as its referenceGuid. Other clients would not be
+		// able to perform the rebase for such an operation. To handle this problem
+		// we have to resubmit a version of the operations which has been rebased to
+		// the current remote tip. We already have these rebased versions of the operations
+		// in our localChanges, because we continuously update those to follow the tip.
+		// Therefore our reSubmitCore function searches for the rebased operation in the
+		// localChanges array and submits this up-to-date version instead of the old operation.
+		const rebasedOperation = find(this.localChanges, (op) => op.guid === content.guid);
+
+		if (rebasedOperation) {
+			this.submitLocalMessage(cloneDeep(rebasedOperation), localOpMetadata);
+		} else {
+			// Could this happen or is there a guard that we will never resubmit an already submitted op?
+			console.warn("Resubmitting operation which has already been received back.");
```

Comment:
The later: once the DDS has observed that the pending op has been sequenced by the service (i.e., received it back), it will never be asked to resubmit.